src/client/workflow/serve.ts

```diff
@@ -74,21 +76,38 @@ export const serve = <
    * @returns A promise that resolves to a response.
    */
   return async (request: TRequest) => {
-    const { workflowContext, isFirstInvocation } =
-      await WorkflowContext.createContext<TInitialPayload>(request, client, initialPayloadParser);
+    const callReturnCheck = await handleCallReturn(request, client);
```

Comment:
Let's document this and make intentions clearer. Why'd we be handlingCallReturn? If those are specifically for 3rd party calls maybe we can come up with a better name?

Reply:
renamed handleCallReturn as handleThirdPartyCallResult and explained how third party calls work in handleThirdPartyCallResult docs